### PR TITLE
Check is lesson done

### LINKED ISSUE #489: Add endpoint CheckIsLessonDone
We can add a lesson, but in fact, it can be canceled. We should have a possibility to check whether the teacher taught the lesson or not.

### PR DESCRIPTION
**Pull request by Serwlashev**
Now we can check was lesson done or not. In case the lesson was done we get response 200 and true. If the lesson not found we get NotFoundException. If the lesson was canceled or did not done we get false and code 200

Screenshot_12
Screenshot_6

### CODE REVIEW

File: CharlieBackend.Business/Services/LessonService.cs [hunk 1]
```diff
@@ -287,5 +289,22 @@ public async Task<Result<LessonDto>> GetLessonByIdAsync(long lessonId)
 
             return Result<LessonDto>.GetSuccess(lessonDto);
         }
+
+        public async Task<Result<bool>> IsLessonDoneAsync(long id)
```
Comment (mahotskuy) [inline CharlieBackend.Business/Services/LessonService.cs]: Return Task<bool> here

File: CharlieBackend.Api/Middlewares/ExceptionHandleMiddleware.cs [hunk 2]
```diff
@@ -52,6 +52,9 @@ private Task HandleExceptionAsync(HttpContext context, Exception exception)
                 case nameof(EntityValidationException):
                     context.Response.StatusCode = 400;
                     break;
+                case nameof(LessonNotDoneException):
+                    context.Response.StatusCode = 403;
```
Comment (mahotskuy) [inline CharlieBackend.Api/Middlewares/ExceptionHandleMiddleware.cs]: return 400 error.
Add this to appropriate case

Comment (Serwlashev) [reply CharlieBackend.Api/Middlewares/ExceptionHandleMiddleware.cs]: Fixed
Screenshot_14

File: CharlieBackend.Business/Services/LessonService.cs (hunk 1 above)
Comment (Serwlashev) [reply CharlieBackend.Business/Services/LessonService.cs]: Fixed
Screenshot_15

File: CharlieBackend.Business/Services/LessonService.cs [hunk 3]
```diff
@@ -287,5 +289,22 @@ public async Task<Result<LessonDto>> GetLessonByIdAsync(long lessonId)
...
+            {
+                throw new NotFoundException("Given lesson not found");
+            }
+
+            if(lesson.Visits.Where(visit => visit.Presence == true).Any())
+            {
+                return true;
+            }
+
+            throw new LessonNotDoneException("The lesson was not done");
```
Comment (mahotskuy) [inline CharlieBackend.Business/Services/LessonService.cs]: Return bool value here and remove LessonNotDoneException as we don't use it for now.

Comment (Serwlashev) [reply CharlieBackend.Business/Services/LessonService.cs]: Done
Screenshot_6

**Review by mahotskuy — APPROVED** [review]

**Review by zelya2021 — APPROVED** [review]

**Review by apyura — APPROVED** [review]

**Pull request merged**